### PR TITLE
[reland] switch mean to use reduction linear

### PR DESCRIPTION
**Pull request by wanchaol**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #97996

mean is actually a reduction linear formula if the final reduction
is partial sum (which currently is), so switching to use that instead

### CODE REVIEW

**Review by XilunWu — APPROVED** [review]
Thx for merge mean rule! One question.

File: torch/distributed/_tensor/ops/math_ops.py [hunk 1]
```diff
@@ -36,6 +36,9 @@ def default_reduction_rule(op_schema: OpSchema) -> OutputSharding:
     [
         aten.sum.default,
         aten.sum.dim_IntList,
+        aten.mean.default,
+        aten.mean.dim,
+        aten.mean.out
     ]
 )
 def sum_rule(op_schema: OpSchema) -> OutputSharding:
```
Comment (XilunWu) [inline torch/distributed/_tensor/ops/math_ops.py:44]: since this rule is not only for `sum`, we can rename it to something like `linear_reduction_rule`.

Comment (wanchaol) [reply torch/distributed/_tensor/ops/math_ops.py:44]: sure, will do it later

**Review by yifuwang — APPROVED** [review]
Awesome!

**Comment by wanchaol** [comment]
> @pytorchbot revert -m 'Sorry for reverting your PR, but it fails a test on CPU https://hud.pytorch.org/pytorch/pytorch/commit/1b323b313ce35c03583ece017f928079f4a86882' -c nosignal
> 
> I want to restore distributed CPU tests, but then it's still timeout frequently in turnk AFAICT

@huydhn I couldn't why see the CPU failure correlates to my change though..

**Comment by huydhn** [comment]
> @huydhn I couldn't why see the CPU failure correlates to my change though..

I see. I make a guess because the test starts to fail with your change.  Let me run it with `ciflow/unstable`, and I will reland this if it pass.  I will also do a rebase just to avoid landrace.

**Comment by huydhn** [comment]
I think the same test also fails on the PR, thoughts?

**Comment by wanchaol** [comment]
> I think the same test also fails on the PR, thoughts?

Oh now i see some error related, let me debug a bit and get back, thanks @huydhn!

**Comment by wanchaol** [comment]
> I think the same test also fails on the PR, thoughts?

@huydhn fixed the tests and unstable CI, landing. Thanks for detecting the issue!

**Pull request closed without merging**